lib/chef/resource/locale.rb

```diff
@@ -69,8 +70,12 @@ def lc_all(arg = nil)
         description "Update the system's locale."
         unless up_to_date?
           converge_by "Updating System Locale" do
```

Comment:
This resource needs a bit of modernizing. If we load the current value of :lang then we can use a converge_if_changed to update the value. That will provide us with proper idempotency and allows us to remove the up_to_date? method. That method just becomes the Linux side of the logic that loads the :lang value for current_resource.

Reply:
@tas50 In `up_to_date?` method we are comparing `/etc/locale.conf` content with the new contain Whether any modification is required in `/etc/locale.conf`